For now we have to use one specific version of lxc/lxd that supports
both old and new API. This should be PRed to LXD when we get rid of lxdclient
completely.
https://github.com/lxc/lxd/pull/4343

--- a/github.com/lxc/lxd/client/operations.go
+++ b/github.com/lxc/lxd/client/operations.go
@@ -22,6 +22,11 @@ type Operation struct {
 	chActive chan bool
 }
 
+// NewOperation creates an operation from provided parameters, used for mocking.
+func NewOperation(op api.Operation, r *ProtocolLXD, listener *EventListener, handlerReady bool, handlerLock sync.Mutex, chActive chan bool) (Operation) {
+        return Operation{op, r, listener, handlerReady, handlerLock, chActive}
+}
+
 // AddHandler adds a function to be called whenever an event is received
 func (op *Operation) AddHandler(function func(api.Operation)) (*EventTarget, error) {
 	// Make sure we have a listener setup
@@ -283,6 +288,11 @@ type RemoteOperation struct {
 	err    error
 }
 
+// NewRemoteOperation creates a remote operation from provided parameters, used for mocking.
+func NewRemoteOperation(targetOp *Operation, handlers []func(api.Operation), chDone, chPost chan bool, err error) (RemoteOperation) {
+        return RemoteOperation{targetOp, handlers, chDone, chPost, err}
+}
+
 // AddHandler adds a function to be called whenever an event is received
 func (op *RemoteOperation) AddHandler(function func(api.Operation)) (*EventTarget, error) {
 	var err error
